docs/infrastructure/EKS/main/cluster_masters.tf

```diff
@@ -30,11 +30,46 @@ resource "aws_eks_cluster" "main" {
 
 }
 
+# Get an authentication token to communicate with an EKS cluster, using the "kubernetes" provider.
+data "aws_eks_cluster_auth" "eks_cluster_auth" {
+  name = "${aws_eks_cluster.main.name}"
+}
+
+provider "kubernetes" {
+  host                   = "${aws_eks_cluster.main.endpoint}"
+  cluster_ca_certificate = "${base64decode(aws_eks_cluster.main.certificate_authority.0.data)}"
+  token                  = "${data.aws_eks_cluster_auth.eks_cluster_auth.token}"
+  load_config_file       = false
+  version                = "~> 1.5"
+}
```

Comment:
I'm not sure this would work. The cluster configuration only exists after the cluster was created, but providers are setup when you run `terraform init` (i.e., before the cluster is created). The first run would probably fail, and you'd have to run it second time for it to work.

I think it would be better to provide the correct configmap as an output and run kubectl apply on it, thus avoiding the bootstrap issue. This already exists. Just run:
```
terraform output aws_auth_configmap | kubectl apply -f -
```
after everything is created.

Reply:
I tried, it worked

Reply:
In fact I copied from the internets.. some guy in github did this in his template. It made sense to me and seems to work.

Reply:
If it works, great. Maybe it was something was fixed recently. I seem to recall this type of thing being an issue. Make sure it works on a clean installation, and remove the `.terraform/` to test, if it exists.

Reply:
Done that. It seems like it works.